arch/riscv/Kconfig

```diff
@@ -128,6 +128,24 @@ config GEN_IRQ_VECTOR_TABLE
 config NUM_IRQS
 	int
 
+config CODE_DATA_RELOCATION
+	bool "Support code/data section relocation"
+	depends on SOC_SERIES_RISCV32_IT8XXX2 \
+	  || SOC_RISCV32_LITEX_VEXRISCV \
+	  || SOC_SERIES_GD32VF103 \
+	  || SOC_SERIES_RISCV32_MIV \
+	  || SOC_SERIES_RISCV64_MIV \
+	  || SOC_SERIES_NEORV32 \
+	  || SOC_SERIES_RISCV_SIFIVE_FREEDOM \
+	  || SOC_SERIES_STARFIVE_JH71XX \
+	  || SOC_SERIES_RISCV_TELINK_B91 \
+	  || SOC_SERIES_RISCV_VIRT
```

Comment:
Uhm, this is not particularly scalable and we will 100% forget to keep this updated with new SoCs coming. Can you instead add something like `SOC_SUPPORTS_RELOC` (please, find a better name :) that each SoC can set at SoC defconfig level and make this symbol depending on that instead?

Reply:
Added `ARCH_SUPPORTS_CODE_DATA_RELOCATION` at top-level and made all currently-supporting systems select it.